### PR TITLE
NEW: add trigger for linking and unlinking invoices

### PR DESCRIPTION
**Pull request by altairis-noe**
# NEW add trigger for linking and unlinking invoices
Some external modules might benefit from being able to act upon discount being added to (or removed from) invoices

### CODE REVIEW

File: htdocs/core/class/discount.class.php [hunk 1]
```diff
@@ -4,6 +4,7 @@
  * Copyright (C) 2024      Alexandre Janniaux   <alexandre.janniaux@gmail.com>
  * Copyright (C) 2024       Frédéric France             <frederic.france@free.fr>
  * Copyright (C) 2024		MDW							<mdeweerd@users.noreply.github.com>
+ * Copyright (C) 2024		Noé Cendrier		<noe.cendrier@altairis.fr>
```
Comment (altairis-noe) [inline htdocs/core/class/discount.class.php:7]: @eldy My email address is repelled by by pre-commit, and I don’t remember how we addressed this last the last time 😅

**Comment by eldy** [comment]
I added "noe" int the exception of spellchecker

File: htdocs/core/class/discount.class.php [hunk 2]
```diff
@@ -470,6 +473,14 @@ public function link_to_invoice($rowidline, $rowidinvoice)
 				$this->fk_facture_line = $rowidline;
 				$this->fk_facture = $rowidinvoice;
 			}
+			if (!$notrigger) {
+				// Call trigger
+				$result = $this->call_trigger('DISCOUNT_LINK_TO_INVOICE', $user);
```
Comment (eldy) [inline htdocs/core/class/discount.class.php:478]: Trigger codes must be limited to the 4 CRUD events. Here we modify the discount line, so you can use the code
DISCOUNT_MODIFY

**Comment by altairis-noe** [comment]
> I added "noe" int the exception of spellchecker

Thanks!

File: htdocs/core/class/discount.class.php (hunk 2 above)
Comment (altairis-noe) [reply htdocs/core/class/discount.class.php]: Thanks for the correction!

**Pull request merged**